tests/check_shapes/integration/test_jax.py

```diff
@@ -0,0 +1,194 @@
+# Copyright 2022 The GPflow Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=import-error  # Dependencies might not be installed.
+# pylint: disable=ungrouped-imports
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from time import perf_counter
+from typing import Any, Mapping, Optional
+
+import pytest
+
+from check_shapes import Shape
+from check_shapes import check_shape as cs
+from check_shapes import check_shapes, inherit_check_shapes
+
+from ..utils import assert_has_shape
+
+try:
+    import jax.numpy as jnp
+    import numpy as np
+    from jax import grad, jit, random
+
+    requires_jax = lambda f: f
```

Comment:
Neat trick! Is it worth putting all the requires_* and associated imports in a single file somewhere?